device/globals/item.gd

```diff
@@ -99,7 +99,11 @@ func input(event):
 	if event is InputEventMouseButton || event.is_action("ui_accept"):
 		if event.is_pressed():
 			clicked = true
-			get_tree().call_group_flags(SceneTree.GROUP_CALL_DEFAULT, "game", "clicked", self, event.get_global_position(), event)
+
+			if event.button_index == BUTTON_LEFT:
+				get_tree().call_group_flags(SceneTree.GROUP_CALL_DEFAULT, "game", "clicked", self, event.get_global_position(), event)
+			elif event.button_index == BUTTON_RIGHT:
+				get_tree().call_group_flags(SceneTree.GROUP_CALL_DEFAULT, "game", "right_clicked", self, event.get_global_position(), event)
```

Comment:
What if we introduce the concept of primary and secondary click here and use the button index and the right_mouse_button_action_menu property to check if the primary or secondary button was clicked? `right_clicked()` would be `secondary_clicked()` or something instead and you pass the boolean to those functions and have logic depend on that instead of the button_index. I think that will be more compliant with the future API, and should work better with other input devices as well, I think.

Reply:
I don't really know what the future API is, is that related to the master-branch-of-Godot stuff we saw from Straton earlier?

I think that is definitely the direction to go towards, to decouple the literal event from the idea of the event, but is it relevant in the scope of this PR?

To me it seems like it could be on top of this and have one less PR in the queue.

I'm a bit unsure about what you mean with the right_mouse_button_action_menu being passed; do you mean that the check to see if an action menu should be spawned is a part of the function signature and not a check in the function body?

Reply:
> I don't really know what the future API is, is that related to the master-branch-of-Godot stuff we saw from Straton earlier?

Yeah, I meant the stuff that's been discussed in #44.

> I think that is definitely the direction to go towards, to decouple the literal event from the idea of the event, but is it relevant in the scope of this PR?

Maybe not, but if there's a simpler and more future-proof way to achieve the same thing, I think it's worth discussing before code is merged.

> I'm a bit unsure about what you mean with the right_mouse_button_action_menu being passed; do you mean that the check to see if an action menu should be spawned is a part of the function signature and not a check in the function body?

I was in a bit of a hurry, so some of the stuff I wrote didn't make sense. What I meant was that you'd have something like this in the input function:

    var primary_button = event.button_index == BUTTON_LEFT or (event.button_index == BUTTON_RIGHT and ProjectSettings.get_property("escoria/right_is_primary")
    var position = event.get_global_position()

    get_tree().call_group_flags(SceneTree.GROUP_CALL_DEFAULT, "game", "clicked", self, position, primary_button)

and replace the logic in `clicked()` that depends on `event.button_index` with a check for primary_button.

I realize now that the "right click action menu" setting isn't exactly the same as flipping the buttons though, since you still want to walk with left-click, right?

So doing what I'm thinking of cleanly also require checking whether the click happened on an item or the background, and probably some other things, so I agree that it's outside of the scope of this PR. When that's in place though, there won't be a need for a second clicked menu, I think, but maybe it's a good idea to have one in the meantime, even though it duplicates a lot of code.

What do you think, @StraToN?